kphp-inspector/src/file_searcher.js

```diff
@@ -38,42 +39,164 @@ function doesFileContainFunction(fullFileName) {
+
+  let [queryParts, longestPart] = queryToParts(query);
+  let files = findClassFilesWithNameLike(longestPart);
+  utils.debug(`Found files: ${files.length === 0 ? 'empty' : files.join(', ')}`);
+
+  return files.filter(filename => filename.endsWith('.h') && doesFileSatisfyQ(filename, queryParts));
+}
+
+/**
+ * Convert search query into parts and looking for the longest.
+ * For example
+ *   "ClassName::method" -> parts: ['classname','method'], longest: 'classname'
+ *   "VK Feed something" -> parts: ['vk','feed','something'], longest: 'something'
+ *
+ * Returns the parts in first element and the longest part in second.
+ *
+ * @param {string} query
+ * @returns {[string[], string]}
+ */
+function queryToParts(query) {
+  let queryParts = splitUserSearchStr(query);
+  if (queryParts.length === 0) {
+    return [[], ''];
+  }
+
+  utils.debug(`Query parts: ${queryParts.join(', ')}`);
+
+  let longestQueryLen = Math.max(...queryParts.map(p => p.length));
+  let longestPart = queryParts.find(p => p.length === longestQueryLen);
+
+  return [queryParts, longestPart];
+}
+
+/**
+ * Searches for files for classes.
+ * @see findFilesWithNameLike
+ *
+ * @param {string} query Search string from console
+ * @return {string[]}    An array of found files
+ */
+function findClassFilesWithNameLike(query) {
+  return findFilesWithNameLike(query, `${env.RUN_ARGV.KPHP_COMPILED_ROOT}/cl`, false);
+}
+
+/**
+ * Searches for files for functions with additional filtering.
+ * @see findFilesWithNameLike
+ *
+ * @param {string} query         Search string from console
+ * @param {boolean} searchPrefix Specifies whether to search for a prefix or a substring
+ * @return {string[]}            An array of found files
+ */
+function findFunctionFilesWithNameLike(query, searchPrefix = false) {
+  return findFilesWithNameLike(query, env.RUN_ARGV.KPHP_COMPILED_ROOT, searchPrefix)
+    .filter((filename, _, output) =>
+      // strip out duplicates and strange files: leave only .cpp and .h, also filter out .h files of classes
+      filename.endsWith('.cpp') ||
+      (
+        filename.endsWith('.h') &&
+        !output.includes(filename.substr(0, filename.length - 2) + '.cpp') &&
+        !filename.includes('cl/C@')
+      ),
+    );
+}
+
+/**
+ * Function searches for files in `searchFolder` that have the `query` substring in the name.
+ *
+ * If `searchPrefix` is true, then the search will search for files with the `query` prefix.
+ *
+ * @param {string} query         Search string from console
+ * @param {string} searchFolder  Folder among the files to be searched
+ * @param {boolean} searchPrefix Specifies whether to search for a prefix or a substring
+ * @return {string[]}            An array of found files
+ */
+function findFilesWithNameLike(query, searchFolder = env.RUN_ARGV.KPHP_COMPILED_ROOT, searchPrefix = false) {
+  let findQuery;
+  if (searchPrefix) {
+    findQuery = `${query}*`;
+  } else {
+    findQuery = `*${query}*`;
+  }
```

Comment:
Could be rewritten with a ternary right in declaration

Reply:
Fixed.